operator/pkg/metrics/monitoring.go

```diff
@@ -0,0 +1,240 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"istio.io/pkg/monitoring"
+)
+
+// OperatorVersionLabel describes version of running binary
+var OperatorVersionLabel = monitoring.MustCreateLabel("v")
+
+// MergeErrorLabel describes the type of merge error
+var MergeErrorLabel = monitoring.MustCreateLabel("error_type")
+
+// PathLabel describes translated path
+var PathLabel = monitoring.MustCreateLabel("path")
+
+// RenderErrorLabel describes the type of the error while rendering
+var RenderErrorLabel = monitoring.MustCreateLabel("render_error")
+
+var (
+	// CRFetchErrorReasonLabel describes the reason/HTTP code
+	// for failing to fetch CR
+	CRFetchErrorReasonLabel = monitoring.MustCreateLabel("reason")
+
+	// CRNamespacedNameLabel describes the name with namespace
+	CRNamespacedNameLabel = monitoring.MustCreateLabel("name")
+)
+
+var (
+	// ComponentNameLabel represents istio component name - like
+	// core, pilot, istio-cni etc.
+	ComponentNameLabel = monitoring.MustCreateLabel("component")
+
+	// ResourceKindLabel indicates the kind of resource owned
+	// or created or updated or deleted or pruned by operator
+	ResourceKindLabel = monitoring.MustCreateLabel("kind")
+
+	// CRRevisionLabel indicates the revision in the CR
+	CRRevisionLabel = monitoring.MustCreateLabel("revision")
+)
+
+// MergeErrorType describes the class of errors that could
+// occur while merging profile, user supplied YAML, values
+// overridden by --set and so on
+type MergeErrorType string
+
+const (
+	// CannotFindProfileError occurs when profile cannot be found
+	CannotFindProfileError MergeErrorType = "fetch_profile"
```

Comment:
i'd recommend making the error name == text (can be snake case). having two description labels can be confusing eg. cannot_find_profile_error

Reply:
How about this convention uniformly? Snake case in Golang seems a bit weird except in places where we have generated protobuf - something like `InstallStatus_Status`
```go
CannotFindProfileError = "cannot_find_profile"
```
Similar but not exactly the same - https://golang.org/src/net/http/status.go